docusaurus/docs/iOS/controllers/channels.md

```diff
@@ -0,0 +1,204 @@
+---
+title: Channel Controllers
+---
+
+## ChannelListController
+
+`ChatChannelListController` allows you to observe a list of chat channels based on the provided query and to paginate channels.
+
+### ChannelList Delegate
+
+Classes that conform to the `ChatChannelListControllerDelegate` protocol will receive changes to the queried list of channels.
+
+```swift
+func controller(
+    _ controller: ChatChannelListController,
+    didChangeChannels changes: [ListChange<ChatChannel>]
+)
+```
+
+## ChannelController
+
+`ChatChannelController` allows you to observe and mutate data for one channel.
+
+### Channel Delegate
+
+Classes that conform to the `ChatChannelControllerDelegate` protocol will receive changes to channel data, members, messages and currently typing users.
+
+```swift
+func channelController(
+    _ channelController: ChatChannelController,
+    didUpdateChannel channel: EntityChange<ChatChannel>
+) {}
+
+func channelController(
+    _ channelController: ChatChannelController,
+    didUpdateMessages changes: [ListChange<ChatMessage>]
+) {}
+
+func channelController(
+    _ channelController: ChatChannelController,
+    didChangeTypingUsers typingUsers: Set<ChatUser>
+) {}
+
+func channelController(_ channelController: ChatChannelController, didReceiveMemberEvent: MemberEvent) {}
+```
```

Comment:
These functions seem like they do not require each other to function. If they do not, we should break this code block up into separate units with some prefacing content about what they do to improve scan-ability.

Reply:
this is a self-explanatory way to explain the delegate protocol